src/editor/scripting/builtin.cc

```diff
@@ -0,0 +1,2225 @@
+      new FunctionBase("print",
+                       VariableType(VariableTypeID::Nil), // call on
+                       VariableType(VariableTypeID::Nil), // returns
+                       {std::make_pair("text", VariableType(VariableTypeID::String))})),
+   new BuiltinFunctionInfo(
+      "random_1",
+      /** TRANSLATORS: max is the name of a function parameter */
+      []() { return _("Returns a random value between 1 and max."); },
+      new FunctionBase("math.random",
+                       VariableType(VariableTypeID::Nil), // call on
+                       VariableType(VariableTypeID::Integer), // returns
+                       {std::make_pair("max", VariableType(VariableTypeID::Integer))})),
+   new BuiltinFunctionInfo(
+      "random_2",
+      /** TRANSLATORS: min and max are names of function parameters */
+      []() { return _("Returns a random value between min and max."); },
+      new FunctionBase("math.random",
+                       VariableType(VariableTypeID::Nil), // call on
+                       VariableType(VariableTypeID::Integer), // returns
+                       {std::make_pair("min", VariableType(VariableTypeID::Integer)),
+                        std::make_pair("max", VariableType(VariableTypeID::Integer))})),
+
+	// Shipped Lua functions
+
+   new BuiltinFunctionInfo(
+      "sleep",
+      []() { return _("Pauses the current thread for the given number of milliseconds."); },
+      new FunctionBase("sleep",
+                       VariableType(VariableTypeID::Nil), // call on
+                       VariableType(VariableTypeID::Nil), // returns
+                       {std::make_pair("milliseconds", VariableType(VariableTypeID::Integer))}),
+      "scripting/coroutine.lua"),
+   new BuiltinFunctionInfo(
+      "wake_me",
+      []() { return _("Pauses the current thread and resumes it at the given gametime (in milliseconds)."); },
+      new FunctionBase("wake_me",
+                       VariableType(VariableTypeID::Nil), // call on
+                       VariableType(VariableTypeID::Nil), // returns
+                       {std::make_pair("time", VariableType(VariableTypeID::Integer))}),
+      "scripting/coroutine.lua"),
+
+   // TODO(Nordfriese): This function can take varargs
+   new BuiltinFunctionInfo("array_combine", []() { return
+		_("Concatenates the given arrays into a single array."); },
+		new FunctionBase("array_combine", VariableType(VariableTypeID::Nil), // call on
+   		VariableType(VariableType(VariableTypeID::Integer), VariableType(VariableTypeID::Any)), // returns
+                       {std::make_pair("array1", VariableType(VariableType(VariableTypeID::Integer),
+                       VariableType(VariableTypeID::Any))),
+                       std::make_pair("array2", VariableType(VariableType(VariableTypeID::Integer),
+                       VariableType(VariableTypeID::Any)))}),
+      "scripting/table.lua"),
+
+   new BuiltinFunctionInfo(
```

Comment:
We should have some automated tests here to ensure that the function definitions won't diverge between this implementation and the actual Lua script files. Will be some work and should be done in a follow-up branch.